File: VB/CustomReportControlSample/CustomLabel/CustomLabel.vb
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.Utils.Serializing

Namespace CustomReportControlSample
	Public Class CustomLabel
		Inherits XRLabel

		<Browsable(True), Bindable(False), Category("Data"), XtraSerializableProperty, DefaultValue(""), DesignerSerializationVisibility(DesignerSerializationVisibility.Visible), EditorBrowsable(EditorBrowsableState.Always)>
		Public Property CustomNumber() As String

		<Browsable(True), Bindable(True), Category("Data"), XtraSerializableProperty, DefaultValue(""), DesignerSerializationVisibility(DesignerSerializationVisibility.Visible), EditorBrowsable(EditorBrowsableState.Always)>
		Public Property StringData() As String

		Protected Overrides Sub PutStateToBrick(ByVal brick As DevExpress.XtraPrinting.VisualBrick, ByVal ps As DevExpress.XtraPrinting.PrintingSystemBase)
			MyBase.PutStateToBrick(brick, ps)
			brick.Text = Me.StringData
		End Sub
	End Class
End Namespace

File: CS/CustomReportControlSample/CustomLabel/CustomLabel.cs
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.Utils.Serializing;

namespace CustomReportControlSample
{
    public class CustomLabel : XRLabel {
        [Browsable(true), Bindable(false), Category("Data")]
        [XtraSerializableProperty]
        [DefaultValue("")]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Visible),
        EditorBrowsable(EditorBrowsableState.Always)]
        public string CustomNumber { get; set; }

        [Browsable(true), Bindable(true), Category("Data")]
        [XtraSerializableProperty]
        [DefaultValue("")]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Visible),
        EditorBrowsable(EditorBrowsableState.Always)]
        public string StringData { get; set; }

        protected override void PutStateToBrick(DevExpress.XtraPrinting.VisualBrick brick, 
                                                DevExpress.XtraPrinting.PrintingSystemBase ps) {
            base.PutStateToBrick(brick, ps);
            brick.Text = StringData ?? CustomNumber;
        }
    }
}
